### DISCUSSION TITLE
Where are the logs?

### DISCUSSION THREAD

**Discussion by engineer-plus-plus** [objective-see/LuLu#602] [Q&A] [answered]
Where can I find the logs of network connections managed by LuLu?  Which ones were allowed, which ones reject, etc.  Logs that one can use for forensics and such.  Dose Lulu keep those?

**Comment by objective-see** [comment, marked as answer]
LuLu uses the macOS's unified logging mechanism. 

You can watch the log messages in real time in the terminal via:
`log stream --level debug --predicate="subsystem='com.objective-see.lulu'"`

You can extract persistent log messages via:
`log show --predicate="subsystem='com.objective-see.lulu'"`

Note: Most of LuLu's log messages are debug msgs that macOS doesn't save, however user responses, etc are saved.
